### PR TITLE
Guide check for contribution

### PR DESCRIPTION
**Pull request by edwinjosegeorge**
Enable the checks for code quality locally.

  1.  Add pre-commit to run git hooks for flake8, black and mypy.
  2.  Fix: async pytest requires pytest-asyncio library to handle coroutines.

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -427,6 +427,17 @@ files = [
 [package.dependencies]
 pycparser = "*"
 
+[[package]]
+name = "cfgv"
```
Comment (ghost) [inline poetry.lock:431]: what is this? @edwinjosegeorge

**Comment by ghost** [comment]
hey @edwinjosegeorge we already have this with pre-commit hooks 

I don't think you need to run 

```
poetry run flake8
poetry run pytest .
```


pre-commit should take care of this - https://github.com/BerriAI/litellm/blob/9b7db7ea47156b74a42fac9c0af2cc634943ac8b/.pre-commit-config.yaml#L4

File: poetry.lock (hunk 1 above)
Comment (edwinjosegeorge) [reply poetry.lock:431]: These packages was install as the dependencies of `pre-commit` package under `dev` group. 

image

**Comment by edwinjosegeorge** [comment]
Hi @krrishdholakia

Thanks for the quick response ❤️ .  Upon cloning the repository, I found that there was no mention of `pre-commit` as a guide but I found the config file. The config file does runs `black`, `flake8` and `mypy`, but does not run 'pytest'. Installing and configuring `pre-commit` will surely run these, but not the test files. Maybe we can tag `black`, `flake8` on commit hook and `mypy`, 'pytest' on push hook. 

I also found couple of libraries that is imported, but not used. Running linters `pylint` can catch a lot of inconsistency improving the code quality.

**Comment by edwinjosegeorge** [comment]
closing as outdated

**Pull request closed without merging**